specutils/spectra/spectrum1d.py

```diff
@@ -673,13 +673,19 @@ def radial_velocity(self, val):
 
     def __add__(self, other):
         if not isinstance(other, (NDCube, u.Quantity)):
-            other = u.Quantity(other, unit=self.unit)
+            try:
+                other = u.Quantity(other, unit=self.unit)
+            except TypeError:
+                return NotImplemented
```

Comment:
You don't really need the `__add__` for specreduce, right? I only found `__rsub__` in specreduce.